samcli/local/docker/lambda_build_container.py

```diff
@@ -249,21 +295,27 @@ def _convert_to_container_dirs(host_paths_to_convert, host_to_container_path_map
         return result
 
     @staticmethod
-    def _get_image(runtime, architecture):
+    def _get_image(runtime, architecture, build_method: Optional[str] = None):
         """
         Parameters
         ----------
         runtime : str
             Name of the Lambda runtime
         architecture : str
             Architecture type either 'x86_64' or 'arm64
+        build_method : str
+            The build method specified in metadata
 
         Returns
         -------
         str
             valid image name
         """
-        return f"{LambdaBuildContainer._IMAGE_URI_PREFIX}-{runtime}:" + LambdaBuildContainer.get_image_tag(architecture)
+        # a list of build methods need to get image based on build method itself instead of runtime
+        get_image_from_build_method = ["dotnet7"]
```

Comment:
Can we move this logic to before the calling this method? Seems weird to put this here and just override the runtime. Also hardcodes specific `dotnet7` things in "random" places. Can we make this more abstract to support similar things for future cases?

For example, If `build_method` is provided, we use that over the runtime directly?

Reply:
Done. I changed the name to `specified_workflow` to be more consistent with `workflow_config.py`